src/client/datascience/insidersNativeNotebookSurveyBanner.ts

```diff
@@ -0,0 +1,133 @@
+import { Experiments } from '../common/experiments/groups';
+import { IBrowserService, IExperimentService, IJupyterExtensionBanner, IPersistentStateFactory } from '../common/types';
+import * as localize from '../common/utils/localize';
+import { INotebookEditorProvider } from './types';
+
+export enum InsidersNotebookSurveyStateKeys {
+    ShowBanner = 'ShowInsidersNotebookSurveyBanner',
+    OpenNotebookCount = 'DS_InsidersNotebookOpenNotebookCount',
+    ExecutionCount = 'DS_InsidersNotebookExecutionCount'
+}
+
+enum DSSurveyLabelIndex {
+    Yes,
+    No
+}
+
+const NotebookOpenThreshold = 5;
+const NotebookExecutionThreshold = 100;
+
+@injectable()
+export class InsidersNativeNotebooksSurveyBanner implements IJupyterExtensionBanner {
+    private get enabled(): boolean {
+        return (
+            this.persistentState.createGlobalPersistentState<boolean>(InsidersNotebookSurveyStateKeys.ShowBanner, true)
+                .value && env.uiKind !== UIKind?.Web
+        );
+    }
+    private disabledInCurrentSession: boolean = false;
+    private bannerMessage: string = localize.InsidersNativeNotebooksSurveyBanner.bannerMessage();
+    private bannerLabels: string[] = [
+        localize.DataScienceSurveyBanner.bannerLabelYes(),
+        localize.DataScienceSurveyBanner.bannerLabelNo()
+    ];
+    private readonly surveyLink: string;
+
+    constructor(
+        @inject(IApplicationShell) private appShell: IApplicationShell,
+        @inject(IPersistentStateFactory) private persistentState: IPersistentStateFactory,
+        @inject(IBrowserService) private browserService: IBrowserService,
+        @inject(INotebookEditorProvider) editorProvider: INotebookEditorProvider,
+        @inject(IExperimentService) private experimentService: IExperimentService,
+        @inject(IApplicationEnvironment) private applicationEnvironment: IApplicationEnvironment,
+        surveyLink: string = 'https://aka.ms/vscjupyternb'
+    ) {
+        this.surveyLink = surveyLink;
+        editorProvider.onDidOpenNotebookEditor(this.openedNotebook.bind(this));
+    }
+
+    public async showBanner(): Promise<void> {
+        const executionCount: number = this.getExecutionCount();
+        const notebookCount: number = this.getOpenNotebookCount();
+        const show = await this.shouldShowBanner(executionCount, notebookCount);
+        if (!show) {
+            return;
+        }
+
+        const response = await this.appShell.showInformationMessage(this.bannerMessage, ...this.bannerLabels);
+        switch (response) {
+            case this.bannerLabels[DSSurveyLabelIndex.Yes]: {
+                await this.launchSurvey();
+                // Disable for 6 months
+                await this.disable(6);
+                break;
+            }
+            case this.bannerLabels[DSSurveyLabelIndex.No]: {
+                // Disable for 3 months
+                await this.disable(3);
```

Comment:
Is this part of a spec?
The user said No to taking a survey and we still bring it up again in 3 months.

Reply:
yes

Reply:
Yeah, see #1095